runtime/compiler/control/JITServerCompilationThread.cpp

```diff
@@ -561,6 +561,13 @@ TR::CompilationInfoPerThreadRemote::processEntry(TR_MethodToBeCompiled &entry, J
       // larger sequence number can pass until I increment lastProcessedCriticalSeqNo
       clientSession->getSequencingMonitor()->exit();
 
+      // If class redefinition using HCR extensions occurred, must clear all the caches
+      bool mustClearCaches = std::find(unloadedClasses.begin(), unloadedClasses.end(), reinterpret_cast<TR_OpaqueClassBlock *>(~0)) != unloadedClasses.end();
```

Comment:
Because we need the sequencing monitor, let's move this code up a bit, before releasing it